web/src/settings_components.ts

```diff
@@ -911,9 +911,9 @@ function switching_to_private(
 
 export function save_discard_widget_status_handler(
     $subsection: JQuery,
-    for_realm_default_settings: boolean,
-    sub: StreamSubscription | undefined,
-    group: UserGroup | undefined,
+    for_realm_default_settings: boolean | undefined = false,
```

Comment:
Why is this `| undefined`? Does a caller pass that value explicitly? If so, is that correct?

Reply:
This is wrong!!. Thanks for pointing it out. This can be optional (ie. might not be passed and default value handles it) but not `undefined` . Though there are cases where the caller explicity passes the value -- but it is because the actual intent in those cases is to pass `sub` or `group` fields and the `for_realm_default_settings` is passed just to respect the structure of func call.